core/spec/models/spree/promotion/actions/create_quantity_adjustments_spec.rb

```diff
@@ -25,6 +25,31 @@ module Spree::Promotion::Actions
     describe "#compute_amount" do
       subject { action.compute_amount(line_item) }
 
+      context "with a standard calculator" do
+        let(:calculator) { FactoryBot.create :standard_calculator }
+
+        context "with a line item as a nil" do
+          let(:line_item) { nil }
+
+          before do
+            expect(Spree::Deprecation).to_not receive(:warn).
+              with(/^Spree::Calculator::FlatRate#compute returned 0, it should return a BigDecimal/, any_args)
+          end
+
+          it "returns a 0" do
+            expect(action.compute_amount(line_item)).to eq 0
+          end
```

Comment:
We can simplify this by using the subject via `is_expected` and relying on the return type rather than the warning which could change.

```suggestion
          it { is_expected.to be_a BigDecimal }
          it { is_expected.to eq 0 }
```